### ISSUE TITLE
Clean up os_type use.

### ISSUE THREAD

**Issue by skschneider** [GoogleCloudPlatform/PerfKitBenchmarker#745]
There are a few places where we check the --os_type flag that aren't compatible with the benchmark config "os_type" option. Considering OS type is a property of a VM, we should also look into allowing this to be set separately per VM group.

os_type constant definitions are also repeated in scattered locations, and these should be cleaned up.

**Comment by skschneider** [comment]
After #950 and #952 are addressed, there will only be four ways in which FLAGS.os_type is used:

1) _GetValidBenchmarks in benchmark_sets.py. This restricts the set of benchmarks in a PKB run to those that match the Linux/Windows classification of FLAGS.os_type.
2) RunBenchmarks in pkb.py. This restricts Windows benchmarks so that they can only be run if PKB is also executing on a Windows machine.
3) _CreateBenchmarkRunList in pkb.py. This verifies that if a Linux benchmark is being run, all of its VM groups are configured for Linux VMs, and if a Windows benchmark is being run, all of its VM groups are configured for Windows VMs.
4) ReadStaticVirtualMachineFile in static_virtual_machine.py. This method reads each VM's properties from a config file. It expects a certain set of required properties based on FLAGS.os_type, but os_type is actually one of the properties that can be configured. There is a check to ensure that the os_type of each read configuration matches the Linux/Windows classification of FLAGS.os_type, but the VM os_type need not match FLAGS.os_type if they are both different flavors of Linux. For now, because of the restriction from (1) above of only running either Linux or Windows benchmarks in one PKB run based on FLAGS.os_type, it is also guaranteed that all static VM configurations will match the requirements of the benchmarks that are run. That is, when the BenchmarkSpec calls GetStaticVirtualMachine, it will always get a VM that matches the Linux/Windows state of the current benchmark and VM group.

In summary, the current state of the code is such that one run of PKB can include
A) Linux benchmarks with Linux VMs, or
B) Windows benchmarks with Windows VMs (this is only allowed when running PKB on a Windows machine)
No mixing is allowed. The only way to run Windows benchmarks is via --os_type=windows. If --os_type is anything else, PKB will run Linux benchmarks and expect Linux VMs. Until we decide to change this behavior, I think that's all the work for this issue.
